sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala

```diff
@@ -448,7 +448,12 @@ class SQLQueryTestSuite extends QueryTest with SharedSparkSession {
     listFilesRecursively(new File(inputFilePath)).flatMap { file =>
       val resultFile = file.getAbsolutePath.replace(inputFilePath, goldenFilePath) + ".out"
       val absPath = file.getAbsolutePath
+
+      // Replacing '.' to '_' is an workaround of sbt bug which removes test name prior to the
+      // last dot in JUnitXmlReportPlugin.
+      // Please refer https://github.com/sbt/sbt/issues/2949
       val testCaseName = absPath.stripPrefix(inputFilePath).stripPrefix(File.separator)
+        .replace('.', '_')
```

Comment:
@HeartSaVioR, this way affects individual file testing way. This is a bug in SBT and it affects all tests (see #25630)

Reply:
We should rather find a way to override SBT's builtin XML reporter but looks not easy.

Reply:
Or we should find a way to use Scalatest's XML reporter but that seems not working with Java tests given my rough test.

Reply:
@HyukjinKwon 
Ah OK. Thanks for pointing out. Looks like the code relies on `testCase.name` in some places.

Would it work if we replace the '.' character with '_' just when we pass the name to `test` method? I meant here:

```
// Create a test case to run this case.
test(testCase.name) {
  runTest(testCase)
}
```

Reply:
Ah I just saw your comments after commenting. Yeah it's ideal if sbt fixes the issue but looks like they don't mind.

IMHO we should apply the workaround for now if we create dynamic tests which end up with same name like here. We may not want to address all the occurrence (like single test having '.') as it's still trackable but this case they're all marked as `sql` and not trackable in Jenkins.

Reply:
It affects the test way of selecting individual SQL test files because `--z` option is based upon test name. (See the comments avoid in SQLQueryTestSuite).

Reply:
Yes, we might also need to guide it as well. Assuming we don't have `.` in file name, can we just remove postfix of `.sql` and update javadoc? I hope the change is acceptable, given the change is deterministic and intuitive.

Reply:
For adding context, this is to help tracking the issue related to #25913. `subquery/scalar-subquery/scalar-subquery-select.sql` is failing, but as we know it's just represented as `sql` and bunch of tests are having same name which bothers with weird UI impact in Jenkins page (clicking `sql` in Jenkins page is indeterministic now).